tests/pathmap.test.js

```diff
@@ -0,0 +1,123 @@
+/* global describe, beforeEach, it */
+
+import expect from 'expect';
+
+import Pathmap from '../src/pathmap';
+import { valueOf, BooleanType, ArrayType, NumberType } from '../index';
+
+describe('pathmap', ()=> {
+
+  let pathmap;
+  let LightSwitch;
+  let ref;
+  let id;
+  let hall;
+  let closet;
+
+  let current;
+  beforeEach(()=> {
+    ref = Ref({});
+    LightSwitch = class LightSwitch {
+      hall = Boolean;
+      closet = Boolean;
+    };
+    pathmap = Pathmap(LightSwitch, ref);
+    id = pathmap.get();
+    hall = id.hall;
+    closet = id.closet;
+
+    current = pathmap.get;
+  });
+
+  it('exists', () => {
+    expect(pathmap).toBeDefined();
+  });
+  it('has an id delegate which is represents the microstate at the base path', ()=> {
+    expect(id).toBeInstanceOf(LightSwitch);
+    expect(valueOf(id)).toBe(ref.get());
+  });
+
+  it('has children corresponding to the shit to the substates', ()=> {
```

Comment:
`shit` might not be the right word there.

Reply:
oops :)